File: app/src/main/java/com/kml/searchEngine/internalRoomDatabase/GameDatabase.java
package com.kml.searchEngine.internalRoomDatabase;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Game.class}, version = 1)
public abstract class GameDatabase extends RoomDatabase
{
    private static GameDatabase instance;
    public abstract GameDao gameDao();
    public static synchronized GameDatabase getInstance(Context context)
    {
        if (instance == null)
        {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    GameDatabase.class, "gameDatabase.db")
                    .createFromAsset("databases/gameDatabase.db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/kml/data/internalRoomDatabase/GameDatabase.kt
package com.kml.data.internalRoomDatabase

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.kml.models.Game

@Database(entities = [Game::class], version = 1)
abstract class GameDatabase : RoomDatabase() {
    abstract val gameDao: GameDao

    companion object {
        private var INSTANCE: GameDatabase? = null

        fun getInstance(context: Context): GameDatabase {
            synchronized(this)
            {
                var instance = INSTANCE
                if (instance == null) {

                    instance = Room.databaseBuilder(context.applicationContext,
                            GameDatabase::class.java, "gameDatabase.db")
                            .createFromAsset("databases/gameDatabase.db")
                            .fallbackToDestructiveMigration()
                            .build()

                    INSTANCE = instance
                }

                return instance
            }

        }
    }
}